servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -1094,4 +1120,30 @@ public void reportLostFile(long fileId) throws FileDoesNotExistException {
       return mDependencyMap.getPriorityDependencyList();
     }
   }
+
+  /**
+   * MasterInodeTTL periodic check.
+   */
+  public final class MasterInodeTTLCheckExecutor implements HeartbeatExecutor {
+    @Override
+    public void heartbeat()  {
+      synchronized (mInodeTree) {
+        List<Inode> inodes = mInodeTree.getInodeChildrenRecursive(mInodeTree.getRoot());
+        for (Inode inode : inodes) {
+          if (inode.isFile() && !inode.isPinned()) {
+            InodeFile iFile = (InodeFile) inode;
+            long ttl = iFile.getTTL();
+            if (ttl > 0
+                && System.currentTimeMillis() - iFile.getCreationTimeMs() > ttl * 1000) {
+              try {
+                mInodeTree.deleteInode(iFile);
```

Comment:
The deleted files should be journaled. Also, will you be able to re-use deleteInodeInternal()?